src/Shared/Helpers/PackageDownloader.cs

```diff
@@ -57,7 +57,7 @@ public PackageDownloader(PackageURL purl, string destinationDir = null, bool doC
             this.destinationDirectory = string.IsNullOrEmpty(destinationDir) ? 
                 Path.Combine(Path.GetTempPath(), Path.GetRandomFileName()) : destinationDir;
 
-            this.packageManager = this.GetPackageManager(purl, this.destinationDirectory);
+            this.packageManager = ProjectManagerFactory.CreateProjectManager(purl, this.destinationDirectory);
```

Comment:
This throws but you don't catch it.  See other comments regarding use of factory patterns.

Reply:
will fix